### PR TITLE
fix(Dropdown) narrator announcing hidden items as selectable

### LINKED ISSUE #23027: [Bug]: Narrator announces incorrect items count in dropdown
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Window 11, Browser- Chrome
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/rana-khell/pen/ZErLvqR

## Actual Behavior
When focus lands on the first option in the callout, the narrator announces:
"Grape 4 of 7, non selected"
image

## Expected Behavior
When focus lands on the first option in the callout, the narrator should announce:
"Grape 1 of 4, non selected"
because four options are displayed.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by marcosmoura**
## Current Behavior
Currently, the V8 Dropdown component take into consideration all elements rendered into DOM to narrate the amount of options in the listbox, which leads to confusion when there are `hidden` options.

## New Behavior
Fix the component by not including the hidden options into the aria-posinset and aria-setsize. Also, setting the data-focusable to false.

## Related Issue(s)
Fixes #23027

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d8997f68c68e6c71a09f9db87ec54aff9cd9a651'>d8997f68c68e6c71a09f9db87ec54aff9cd9a651</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts [hunk 1]
```diff
@@ -14,24 +14,32 @@ import type { IDropdownOption } from '../Dropdown.types';
 export class DropdownSizePosCache {
   private _cachedOptions: IDropdownOption[];
   private _displayOnlyOptionsCache: number[];
```
Comment (marcosmoura) [inline packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts:16]: - _displayOnlyOptionsCache tracks the items that are not focusable: headers and dividers
- _notSelectableOptionsCache caches all non-selectable option: headers and dividers but also hidden options.

File: packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts [hunk 2]
```diff
@@ -62,14 +70,18 @@ export class DropdownSizePosCache {
     // we could possibly memoize this too but this should be good enough, most of the time (the expectation is that
     // when you have a lot of options, the selectable options will heavily dominate over the non-selectable options.
     let offset = 0;
-    while (index > this._displayOnlyOptionsCache[offset]) {
+    while (index > this._notSelectableOptionsCache[offset]) {
```
Comment (marcosmoura) [inline packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts:73]: Ignore all non-selectable options, not only Headers and Dividers but also `hidden` options.

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/BaseButton-mount_base.html">794</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/BaseButton-mount.html">790</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Breadcrumb-mount_base.html">2321</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Breadcrumb-mount.html">2290</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Checkbox-mount_base.html">1290</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Checkbox-mount.html">1256</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/CheckboxBase-mount_base.html">1067</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/CheckboxBase-mount.html">1059</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ChoiceGroup-mount_base.html">3936</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ChoiceGroup-mount.html">3903</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ComboBox-mount_base.html">773</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ComboBox-mount.html">766</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/CommandBar-mount_base.html">8828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/CommandBar-mount.html">8787</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ContextualMenu-mount_base.html">9277</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ContextualMenu-mount.html">9276</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DefaultButton-mount_base.html">954</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DefaultButton-mount.html">965</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRow-mount_base.html">3135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRow-mount.html">3078</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRowFast-mount_base.html">3104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRowFast-mount.html">2991</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRowNoStyles-mount_base.html">2870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DetailsRowNoStyles-mount.html">2867</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Dialog-mount_base.html">2387</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Dialog-mount.html">2334</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DocumentCardTitle-mount_base.html">153</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/DocumentCardTitle-mount.html">158</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Dropdown-mount_base.html">2827</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Dropdown-mount.html">2840</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/FocusTrapZone-mount_base.html">1612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/FocusTrapZone-mount.html">1569</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/FocusZone-mount_base.html">1552</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/FocusZone-mount.html">1540</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/IconButton-mount_base.html">1444</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/IconButton-mount.html">1408</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Label-mount_base.html">322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Label-mount.html">324</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Layer-mount_base.html">2550</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Layer-mount.html">2523</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Link-mount_base.html">415</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Link-mount.html">436</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/MenuButton-mount_base.html">1208</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/MenuButton-mount.html">1220</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/MessageBar-mount_base.html">1833</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/MessageBar-mount.html">1897</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Nav-mount_base.html">2675</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Nav-mount.html">2642</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/OverflowSet-mount_base.html">1003</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/OverflowSet-mount.html">1017</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Panel-mount_base.html">1866</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Panel-mount.html">1817</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Persona-mount_base.html">867</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Persona-mount.html">860</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Pivot-mount_base.html">1121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Pivot-mount.html">1133</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/PrimaryButton-mount_base.html">1105</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/PrimaryButton-mount.html">1086</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Rating-mount_base.html">6725</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Rating-mount.html">6641</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SearchBox-mount_base.html">1139</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SearchBox-mount.html">1125</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Shimmer-mount_base.html">2235</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Shimmer-mount.html">2198</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Slider-mount_base.html">1705</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Slider-mount.html">1650</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SpinButton-mount_base.html">4015</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SpinButton-mount.html">3968</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Spinner-mount_base.html">406</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Spinner-mount.html">405</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SplitButton-mount_base.html">2496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SplitButton-mount.html">2484</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Stack-mount_base.html">478</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Stack-mount.html">500</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/StackWithIntrinsicChildren-mount_base.html">1885</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/StackWithIntrinsicChildren-mount.html">1859</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/StackWithTextChildren-mount_base.html">4666</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/StackWithTextChildren-mount.html">4658</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SwatchColorPicker-mount_base.html">9100</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/SwatchColorPicker-mount.html">9057</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TagPicker-mount_base.html">2025</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TagPicker-mount.html">2081</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TeachingBubble-mount_base.html">70339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TeachingBubble-mount.html">70630</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Text-mount_base.html">408</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Text-mount.html">402</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TextField-mount_base.html">1154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/TextField-mount.html">1186</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-mount_base.html">948</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-mount.html">954</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-virtual-rerender_base.html">600</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-virtual-rerender.html">619</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1450</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1450</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Toggle-mount_base.html">710</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/Toggle-mount.html">719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/buttonNative-mount_base.html">101</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23755/perf-test/buttonNative-mount.html">111</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Review by loadchange — APPROVED** [review]

**Review by smhigley — APPROVED** [review]

**Pull request merged**